### DISCUSSION TITLE
Import an Image-based network

### DISCUSSION THREAD

**Discussion by JMCarmona10** [PMEAL/OpenPNM#2929] [Q&A]
Hello,
I'm seeking guidance on importing an image-based network from an existing dataset. My question stems from issue #489, where I attempted to recreate the example. However, my network dataset is in .dat format, and I found the code in that example to be outdated.
I've also reviewed the example "Import an Extracted Network and Predict Transport Properties" on the website, but I'm still struggling to understand the process.
Could you please provide instructions or resources on how to properly upload an image-based network? I'm particularly interested in methods that work with current versions of the software and are compatible with .dat file formats.

I also put the files in .dat for better clarity. 

[Network.zip](https://github.com/user-attachments/files/16501358/Network.zip)

**Comment by jgostick** [comment]
A `.dat` file presumably means it is an image?  In this case you need to do a network extraction first, using something like our `snow` algorithm in `porespy`.

**Reply by JMCarmona10** [reply]
the .dat means a network data that are stored in four ASCII ﬁles

**Reply by jgostick** [reply]
4 files makes me think that this is the output from the "maximal ball" code of Blunt's group?  It's the only tool I know that produces 4 files.  If this is the case then you can import it using `openpnm.io.from_statoil`.
